### PR TITLE
feat: [CDS-38023]: Jenkins Build Step

### PR DESCRIPTION
**Pull request by shivamnegi94**
## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)

### CODE REVIEW

**Comment by shivamnegi94** [comment]
trigger messagecheck

File: 120-ng-manager/config.yml [hunk 1]
```diff
@@ -251,7 +251,7 @@ pmsGrpcClientConfig:
   target: localhost:12011
   authority: localhost
 
-shouldConfigureWithPMS: ${SHOULD_CONFIGURE_WITH_PMS:-false}
+shouldConfigureWithPMS: ${SHOULD_CONFIGURE_WITH_PMS:-true}
```
Comment (puthrayaharness) [inline 120-ng-manager/config.yml:254]: Please revert the change

File: 970-ng-commons/src/main/java/io/harness/common/EntityYamlRootNames.java [hunk 2]
```diff
@@ -104,6 +104,7 @@ public class EntityYamlRootNames {
   public static final String SERVERLESS_AWS_LAMBDA_ROLLBACK = "ServerlessAwsLambdaRollback";
   public static final String CLOUDFORMATION_ROLLBACK_STACK_STEP = "RollbackStack";
   public static final String INFRASTRUCTURE = "infrastructure";
+  public static final String JENKINS_BUILD = "jenkinsBuild";
```
Comment (puthrayaharness) [inline 970-ng-commons/src/main/java/io/harness/common/EntityYamlRootNames.java:107]: `JenkinsBuild`

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/CommonStepInfo.java [hunk 3]
```diff
@@ -137,6 +137,15 @@ public class CommonStepInfo {
           .setFeatureFlag(FeatureName.SERVICENOW_CREATE_UPDATE_NG.name())
           .build();
 
+  StepInfo jenkinsBuildStepInfo =
+      StepInfo.newBuilder()
+          .setName("Jenkins Build")
+          .setType(StepSpecTypeConstants.JENKINS_BUILD)
+          .setStepMetaData(StepMetaData.newBuilder().addCategory("Jenkins").addFolderPaths("Jenkins").build())
```
Comment (puthrayaharness) [inline 800-pipeline-service/src/main/java/io/harness/pms/pipeline/CommonStepInfo.java:144]: Please confirm with the Product what should be category and folder paths.

File: 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/PmsStepFilterJsonCreatorV2.java [hunk 4]
```diff
@@ -22,6 +22,9 @@ public class PmsStepFilterJsonCreatorV2 extends GenericStepPMSFilterJsonCreatorV
   public Set<String> getSupportedStepTypes() {
     return Sets.newHashSet(StepSpecTypeConstants.HTTP, StepSpecTypeConstants.JIRA_CREATE,
         StepSpecTypeConstants.JIRA_UPDATE, StepSpecTypeConstants.JIRA_APPROVAL, StepSpecTypeConstants.HARNESS_APPROVAL,
+        StepSpecTypeConstants.SHELL_SCRIPT, StepSpecTypeConstants.SERVICENOW_APPROVAL, StepSpecTypeConstants.BARRIER,
```
Comment (puthrayaharness) [inline 860-orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/PmsStepFilterJsonCreatorV2.java:25]: Please fix this. The change should be mostly Jenkins build. These look to be already present.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 970-ng-commons/src/main/java/io/harness/common/EntityYamlRootNames.java (hunk 2 above)
Comment (shivamnegi94) [reply 970-ng-commons/src/main/java/io/harness/common/EntityYamlRootNames.java]: done

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 10 of 57 files at r1, 1 of 1 files at r2, 7 of 34 files at r3, 72 of 72 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 7 of 72 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: 1 change requests. Waiting on team(s): sto. 

___
*[`860-orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/PmsStepFilterJsonCreatorV2.java` line 28 at r4](https://reviewable.io/reviews/harness/harness-core/33539#-N5hjaH-7pH8puQHz4OY:-N5hjaH02FU73Lsfv04O:b-6qg4r1) ([raw file](https://github.com/harness/harness-core/blob/e78c9f6ff7c3d657d16588be8fce1122d38493a6/860-orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/PmsStepFilterJsonCreatorV2.java#L28)):*
> ```Java
>         StepSpecTypeConstants.SERVICENOW_APPROVAL, StepSpecTypeConstants.BARRIER, StepSpecTypeConstants.POLICY_STEP,
>         StepSpecTypeConstants.SERVICENOW_CREATE, StepSpecTypeConstants.SERVICENOW_UPDATE, StepSpecTypeConstants.QUEUE,
>         StepSpecTypeConstants.SHELL_SCRIPT);
> ```

shell script is already linked here, please remove from here



_Code quote:_
```Java
StepSpecTypeConstants.SHELL_SCRIPT
```
___
*[`860-orchestration-steps/src/main/java/io/harness/serializer/OrchestrationStepsModuleRegistrars.java` line 350 at r4](https://reviewable.io/reviews/harness/harness-core/33539#-N5hjkA40wY0ts2pyNLp:-N5hjkA40wY0ts2pyNLq:b-g7aa9e) ([raw file](https://github.com/harness/harness-core/blob/e78c9f6ff7c3d657d16588be8fce1122d38493a6/860-orchestration-steps/src/main/java/io/harness/serializer/OrchestrationStepsModuleRegistrars.java#L350)):*
> ```Java
>                                            .namespace(SchemaNamespaceConstants.CD)
>                                            .modulesSupported(Arrays.asList(ModuleType.CD, ModuleType.PMS))
>                                            .yamlGroup(YamlGroup.builder().group(StepCategory.STEP.name()).build())
> ```

Why this is needed for jenkins step? or any other change is incorporated in your pr?



_Code quote:_
```Java
                                           .namespace(SchemaNamespaceConstants.CD)
                                           .modulesSupported(Arrays.asList(ModuleType.CD, ModuleType.PMS))
                                           .yamlGroup(YamlGroup.builder().group(StepCategory.STEP.name()).build())
```
___
*[`870-orchestration/src/main/java/io/harness/steps/StepSpecTypeConstants.java` line 38 at r4](https://reviewable.io/reviews/harness/harness-core/33539#-N5hjrfXB7jJ_7k_Sg0Y:-N5hjrfXB7jJ_7k_Sg0Z:b181nta) ([raw file](https://github.com/harness/harness-core/blob/e78c9f6ff7c3d657d16588be8fce1122d38493a6/870-orchestration/src/main/java/io/harness/steps/StepSpecTypeConstants.java#L38)):*
> ```Java
>   String FEATURE_FLAG_STAGE = "FeatureFlag";
>   String POLICY_STEP = "Policy";
>   String JENKINS_BUILD = "JenkinsBuild";
> ```

it should not be in 870 module, it should be in cd



_Code quote:_
```Java
String JENKINS_BUILD = "JenkinsBuild";
```

**Review by archit-harness — APPROVED** [review]
> Reviewed 4 of 72 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto. 

___
*[`860-orchestration-steps/src/main/java/io/harness/serializer/OrchestrationStepsModuleRegistrars.java` line 350 at r4](https://reviewable.io/reviews/harness/harness-core/33539#-N5hjkA40wY0ts2pyNLp:-N5hs_MX1Qs2SFJ5tJmU:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/e78c9f6ff7c3d657d16588be8fce1122d38493a6/860-orchestration-steps/src/main/java/io/harness/serializer/OrchestrationStepsModuleRegistrars.java#L350)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Why this is needed for jenkins step? or any other change is incorporated in your pr?
</blockquote></details>

done
___
*[`870-orchestration/src/main/java/io/harness/steps/StepSpecTypeConstants.java` line 38 at r4](https://reviewable.io/reviews/harness/harness-core/33539#-N5hjrfXB7jJ_7k_Sg0Y:-N5hscT75EzPyI59Dqsk:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/e78c9f6ff7c3d657d16588be8fce1122d38493a6/870-orchestration/src/main/java/io/harness/steps/StepSpecTypeConstants.java#L38)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

it should not be in 870 module, it should be in cd
</blockquote></details>

done

**Review by archit-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto. 

___
*[`860-orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/PmsStepFilterJsonCreatorV2.java` line 28 at r4](https://reviewable.io/reviews/harness/harness-core/33539#-N5hjaH-7pH8puQHz4OY:-N5ht2Tq9vuXKoRQCC1D:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/e78c9f6ff7c3d657d16588be8fce1122d38493a6/860-orchestration-steps/src/main/java/io/harness/plancreator/steps/internal/PmsStepFilterJsonCreatorV2.java#L28)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

shell script is already linked here, please remove from here
</blockquote></details>

done

**Review by shivamnegi94 — COMMENTED** [review]
> Reviewed 10 of 57 files at r1, 1 of 1 files at r2, 7 of 34 files at r3, 57 of 72 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto.

**Review by puthrayaharness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto.

**Review by puthrayaharness — DISMISSED** [review]
> Reviewed 3 of 3 files at r5.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 10 of 57 files at r1, 1 of 1 files at r2, 7 of 34 files at r3, 55 of 72 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539#-:-N5i67VO7s6gG1-CRUM1:bnfp4nl)* status: Waiting on team(s): sto.

**Review by diptiman-adak — APPROVED** [review]
> Reviewed 10 of 57 files at r1, 1 of 1 files at r2, 7 of 34 files at r3, 55 of 72 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto.

**Review by diptiman-adak — APPROVED** [review]
> Reviewed 14 of 72 files at r4, 3 of 3 files at r5.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto.

**Review by prashantsharma04 — DISMISSED** [review]
> Reviewed 10 of 57 files at r1, 1 of 1 files at r2, 7 of 34 files at r3, 55 of 72 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto.

**Review by diptiman-adak — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: Waiting on team(s): sto.

**Review by williamwissemann — APPROVED** [review]
> Reviewed 10 of 57 files at r1, 1 of 1 files at r2, 7 of 34 files at r3, 72 of 72 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: :shipit: complete! ✓

**Review by diptiman-adak — COMMENTED** [review]
> Reviewed 3 of 3 files at r6.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: 29 file(s) to review. 0 of 1 approvals obtained.

**Review by diptiman-adak — COMMENTED** [review]
> Reviewed 3 of 83 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: 29 file(s) to review. 0 of 1 approvals obtained.

**Review by diptiman-adak — APPROVED** [review]
> Reviewed 29 of 29 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: :shipit: complete! ✓

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 3 of 3 files at r6, 29 of 29 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33539)* status: :shipit: complete! ✓

**Pull request merged**